kubernetes/data_source_kubernetes_nodes.go

```diff
@@ -0,0 +1,92 @@
+package kubernetes
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func dataSourceKubernetesNodes() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceKubernetesNodesRead,
+		Schema: map[string]*schema.Schema{
+			"metadata": {
+				Type:        schema.TypeList,
+				Description: "Metadata fields to narrow node selection.",
+				Optional:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"labels": {
+							Type:         schema.TypeMap,
+							Description:  "Select nodes with these labels. More info: http://kubernetes.io/docs/user-guide/labels",
+							Required:     true,
+							Elem:         &schema.Schema{Type: schema.TypeString},
+							ValidateFunc: validateLabels,
+						},
+					},
+				},
+			},
+			"nodes": {
+				Type:        schema.TypeList,
+				Description: "List of nodes in a cluster.",
+				Computed:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
```

Comment:
It would be great to have the entire set of node attributes returned here, instead of just the name. I think that would make the data-source more broadly useful, as well as keeping in line with the rest of the data-sources in the provider.

Reply:
I have added:
- all `metadata` fields
- all `spec` fields (except `configSource` which is deprecated)
- capacity related fields `status.capacity` and `status.allocatable`
- everything in `status.node_info`

The remainder of the fields under status seem to verbose and/or ephemeral to be of use for a data source that returns resources in a list.  It is my opinion that if more detailed information is needed then a `kubernetes_node` data source that retrieves everything for a specific node would make more sense.